src/main/java/com/stripe/StripeClient.java

```diff
@@ -18,18 +19,35 @@ public class StripeClient {
    * instead if you require more complex configuration.
    */
   public StripeClient(String apiKey) {
-    this.responseGetter =
-        new LiveStripeResponseGetter(builder().setApiKey(apiKey).buildOptions(), null);
+    this(new LiveStripeResponseGetter(builder().setApiKey(apiKey).buildOptions(), null));
   }
 
   /**
    * Constructs a StripeClient with a custom StripeResponseGetter.
    *
    * <p>Use this for testing, or advanced use cases where you need to make fundamental changes to
    * how the StripeClient makes requests.
+   *
+   * <p>responseGetter should not be re-used across StripeClient and
+   * ApiResource.setStripeResponseGetter
    */
   public StripeClient(StripeResponseGetter responseGetter) {
     this.responseGetter = responseGetter;
+
+    if (this.responseGetter instanceof LiveStripeResponseGetter) {
+      // Setting usage via mutation instead of making a copy is necessary if we want to support
+      // users to create their own subclasses of LiveStripeResponseGetter to use here.
+      //
+      // Unfortunately, mutation could technically cause some non-obvious behavior in the case
+      // of re-use; e.g.
+      //
+      // var lsrg = new LiveStripeResponseGetter();
+      // ApiResource.setStripeResponseGetter(lsrg);
+      // new StripeClient(lsrg);
+      //
+      // would cause e.g. Customer.create(...) to erroneously report "stripe_client" in usage
+      ((LiveStripeResponseGetter) this.responseGetter).setUsage(Arrays.asList("stripe_client"));
```

Comment:
This makes sense to me. The re-use case seems like a far edge case (where a user has to subclass `LiveStripeResponseGetter`, and is mixing paradigms).

Reply:
Yeah the thing that bothers me is that in our test infra we actually did this... but I still think in real-life code its unlikely.